include/swift/SIL/SILInstructionWorklist.h

```diff
@@ -134,86 +134,88 @@ class SILInstructionWorklist : SILInstructionWorklistBase {
     worklist.clear();
   }
 
-  /// Find usages of \p I and replace them with usages of \p Result.
+  /// Find usages of \p instruction and replace them with usages of \p result.
   ///
-  /// Intended to be called during visitation after \p I has been removed from
-  /// the worklist.
+  /// Intended to be called during visitation after \p instruction has been
+  /// removed from the worklist.
   ///
-  /// \p I the instruction whose usages will be replaced
-  /// \p Result the instruction whose usages will replace \p I
-  bool replaceInstructionWithInstruction(
-      SILInstruction *I, 
-      SILInstruction *Result
+  /// \p instruction the instruction whose usages will be replaced
+  /// \p result the instruction whose usages will replace \p instruction
+  ///
+  /// \return whether the instruction was deleted or modified.
+  bool replaceInstructionWithInstruction(SILInstruction *instruction,
+                                         SILInstruction *result
 #ifndef NDNEBUG
-     , std::string OrigI
+                                         ,
+                                         std::string instructionDescription
 #endif
   ) {
-    if (Result != I) {
-      assert(&*std::prev(SILBasicBlock::iterator(I)) == Result &&
-            "Expected new instruction inserted before existing instruction!");
-
-#ifndef NDEBUG
-      withStream([&](llvm::raw_ostream &stream, StringRef loggingName) {
-        stream << loggingName << ": Old = " << *I      << '\n'
-               << "  "        << "  New = " << *Result << '\n';
+    if (result != instruction) {
+      assert(&*std::prev(SILBasicBlock::iterator(instruction)) == result &&
```

Comment:
Just an FYI for future reference (I am fine if you want to ignore this for now), but the *canonical* way to get an iterator for a SILInstruction is to call the method instruction->getIterator().

Reply:
Noted!

Reply:
Fixed.